botocore/client.py

```diff
@@ -159,6 +163,18 @@ def _conditionally_unregister_fix_s3_host(self, endpoint_url, emitter):
         if endpoint_url is not None:
             emitter.unregister('before-sign.s3', fix_s3_host)
 
+    def _create_s3_redirector(self, client, scoped_config):
+        if client.meta.service_model.service_name != 's3':
+            return
+
+        service_signing_name = client.meta.service_model.signing_name
+        endpoint_bridge = ClientEndpointBridge(
```

Comment:
It is unfortunate that we have to recreate this. Is there any other way we could retrieve the original endpoint bridge so we do not have to make a new one?